handlers/books_handler.go

```diff
@@ -0,0 +1,61 @@
+package handlers
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"main/repository"
+	"net/http"
+	"sort"
+)
+
+func HandleBookRequests(w http.ResponseWriter, r *http.Request) {
+	if r.Method == http.MethodGet {
+		getBooks(w, r)
+	} else if r.Method == http.MethodPost {
+		postBooks(w, r)
+	} else {
+		fmt.Println("supporting only POST and GET requests")
+	}
+}
+
+func getBooks(w http.ResponseWriter, r *http.Request) {
+	booksRepo := repository.NewBookRepo()
+	books, err := booksRepo.GetBooks()
+	if err != nil {
+		println(err)
+		return
+	}
+
+	// sort by date
+	sort.Sort(books)
+
+	json.NewEncoder(w).Encode(books)
+}
+
+func postBooks(w http.ResponseWriter, r *http.Request) {
```

Comment:
Let's talk about POST flow
You retrieve a new list of users
You Marshal it to a list of book's object
You provide it to your repo level where this list is going to be saved to your books parameter